classes/oc/i18n.php

```diff
@@ -76,7 +76,7 @@ public static function initialize($locale = NULL, $charset = 'utf-8', $domain =
          * and force using gettext
          */
         if(defined('LC_ALL'))
-            $locale_res = setlocale(LC_ALL, self::$locale);
+            $locale_res = setlocale(LC_ALL, self::$locale.'.'.self::$charset);
```

Comment:
I think this works fine on Linux but can’t be entirely sure about other OS's.

If not I can’t see how to get localeconv() information with the right charset.